gcore/src/prog.rs

```diff
@@ -109,18 +115,30 @@ pub fn create_program_with_gas(
     gas_limit: u64,
     value: u128,
 ) -> ActorId {
+    let mut inner = [0u8; 56];
+    let fix_len_data = code_hash
+        .iter()
+        .copied()
+        .chain(gas_limit.to_le_bytes())
+        .chain(value.to_le_bytes());
+    populate_from_iterator(&mut inner, fix_len_data);
```

Comment:
Why copy `fix_len_data` to `inner` instead of using `fix_len_data` as a slice?

Reply:
found no way without allocations instead of re implementing that https://crates.io/crates/collect_slice (what actually was done by `populate_from_iterator` function)